pkg/api/methods.go

```diff
@@ -44,3 +48,12 @@ func (t *Taint) ToString() string {
 	}
 	return fmt.Sprintf("%v=%v:%v", t.Key, t.Value, t.Effect)
 }
+
+func (obj *ObjectReference) SetGroupVersionKind(gvk schema.GroupVersionKind) {
+	obj.APIVersion, obj.Kind = gvk.ToAPIVersionAndKind()
+}
```

Comment:
nit: missing blank line

I would slightly prefer not to make a methods.go file which is a dumping ground for methods, what if we kept a ref.go here with just these methods?